instagram-app/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import Comment from './Comment';
+import CommentInput from './CommentInput';
+
+
+class CommentSection extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            comments: props.comments,
+            comment:'',
+            date: props.date
+            
+        };
+    }
+
+    componentDidMount() {
+        let id = this.props.postId;
+        if (localStorage.getItem(id)){
+            this.setState({
+                comments: JSON.parse(localStorage.getItem(this.props.postId))
+            });
+        }
+        else {
+            this.setComments();
+        }
+    }
+
+    componentWillUnmount(){
+        this.setComments();
+    }
+
+    setComments = () =>{
+        localStorage.setItem(
+            this.props.postId,
+            JSON.stringify(this.state.comments)
+        );
```

Comment:
Be aware that the value for `this.props.postId` is actually just the image URL for a post, so you might want to actually add a proper (unique) ID string into your dummy data in order to have a more sensible name.